trustd/src/main.rs

```diff
@@ -25,125 +23,54 @@ pub enum Command {}
+        match self.command {
+            Some(Command::Api(run)) => run.run().await,
+            Some(Command::Db(run)) => run.run().await,
+            None => pm_mode().await,
         }
     }
+}
 
-    async fn run_command(mut self) -> anyhow::Result<ExitCode> {
-        // to keep in scope while running.
-        let mut managed_db = None;
-
-        if matches!(self.database.db_strategy, DbStrategy::Managed) {
-            println!("setting up managed DB");
-            use postgresql_embedded::Settings;
-
-            let current_dir = env::current_dir()?;
-            let work_dir = current_dir.join(".trustify");
-            let db_dir = work_dir.join("postgres");
-            let settings = Settings {
-                username: self.database.username.clone(),
-                password: self.database.password.clone(),
-                temporary: false,
-                installation_dir: db_dir.clone(),
-                ..Default::default()
-            };
-
-            let mut postgresql = PostgreSQL::new(PostgreSQL::default_version(), settings);
-            postgresql.setup().await?;
-            postgresql.start().await?;
-
-            let port = postgresql.settings().port;
-            self.database.port = port;
-            self.creation = CreationMode::Bootstrap;
-
-            managed_db.replace(postgresql);
-
-            println!("postgresql installed under {:?}", db_dir);
-            println!("running on port {}", port);
-        }
-
-        let mut handles = JoinSet::new();
-
-        if self.with_http {
-            let http = trustify_server::Run {
-                database: self.database.clone(),
-                storage: self.storage.clone(),
-                creation: self.creation,
-                devmode: self.devmode,
-                infra: self.infra.clone(),
-                auth: self.auth.clone(),
-                http: self.http.clone(),
-                swagger_ui_oidc: self.swagger_ui_oidc.clone(),
-            };
-
-            handles.spawn_local(http.run());
-        }
-
-        while let Some(result) = handles.join_next().await {
-            match result {
-                Ok(result) => match result {
-                    Ok(_) => {}
-                    Err(err) => {
-                        log::error!("error {:?}", err);
-                    }
-                },
-                Err(err) => {
-                    log::error!("fundamental error {:?}", err);
-                }
-            }
-        }
-
-        Ok(ExitCode::SUCCESS)
+// Project Manager Mode
+async fn pm_mode() -> anyhow::Result<ExitCode> {
+    let Some(Command::Db(mut db)) = Trustd::parse_from(["trustd", "db", "migrate"]).command else {
+        unreachable!()
+    };
+    let postgres = db.start().await?;
+    if postgres.database_exists(&db.database.name).await.is_err() {
+        db.command = db::Command::Create;
     }
+    db.run().await?;
+
+    let api = Trustd::parse_from([
```

Comment:
Interesting. I do love it, because it mirrors what the user would do. But I agree it'd be nice for the compiler to complain if we subsequently change the CLI.

Interestingly, my first pass was as you suggest, but I found myself wanting to implement `Default` for things that are already kind of implemented in the form of clap annotations that I didn't want to replicate. And I wanted to take advantage of that clap config, hence `parse_from`.

It's definitely a tradeoff, but for something like "PM mode", I was ok with it. Better than ok, actually, but if you insist...